### PR TITLE
MINOR: Remove o.a.kafka.common.utils.Base64 and IS_JAVA8_COMPATIBLE

### PR DESCRIPTION
**Pull request by ijuma**
We no longer need them since we now require Java 8.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ijuma** [comment]
@lindong28, does this look OK to you?

**Comment by lindong28** [comment]
@ijuma Sure! It seems that the last commit `e70a191d3038e00790aa95fbd1e16e78c32b79a4` in trunk caused `./gradlew checkstyleMain checkstyleTest` to fail. It prevents this and future PR from running tests. 

Looking at that patch, the check style failure seems to be due to the following change. Can we remove the log.isDebugEnabled() check?

```
if (log.isDebugEnabled()) {
  log.debug("{} attempting protocol downgrade and then retry.", this);
}
```

**Comment by ijuma** [comment]
Yes, I pushed https://github.com/apache/kafka/commit/193c77968251a06b2393d1d286425a4f4ec97d10 shortly after. I have also rebased this PR.

File: clients/src/main/java/org/apache/kafka/common/utils/Java.java [hunk 1]
```diff
@@ -38,7 +38,6 @@ static Version parseVersion(String versionString) {
 
     // Having these as static final provides the best opportunity for compilar optimization
     public static final boolean IS_JAVA9_COMPATIBLE = VERSION.isJava9Compatible();
-    public static final boolean IS_JAVA8_COMPATIBLE = VERSION.isJava8Compatible();
```
Comment (andrasbeni) [inline clients/src/main/java/org/apache/kafka/common/utils/Java.java:41]: @ijuma For me it seems `isJava8Compatible()` could also be removed (with all its test usages)

Comment (ijuma) [reply clients/src/main/java/org/apache/kafka/common/utils/Java.java:41]: Done.

**Review by andrasbeni — APPROVED** [review]
+1

File: core/src/test/scala/unit/kafka/utils/PasswordEncoderTest.scala [hunk 2]
```diff
@@ -109,11 +109,9 @@ class PasswordEncoderTest {
     verifyEncodeDecode(keyFactoryAlg = None, "AES/CFB/PKCS5Padding", keyLength = 128)
```
Comment (omkreddy) [inline core/src/test/scala/unit/kafka/utils/PasswordEncoderTest.scala:109]: nit: unused import

Comment (lindong28) [reply core/src/test/scala/unit/kafka/utils/PasswordEncoderTest.scala:109]: Thanks for catching this. I removed this line before merging the PR.

**Comment by lindong28** [comment]
@ijuma Thanks! LGTM. Merged to trunk.

**Pull request closed without merging**